pkg/generate/app/app.go

```diff
@@ -431,8 +432,13 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
 		r.Name = name
 	}
 
-	selector := map[string]string{
-		"deploymentconfig": r.Name,
+	var selector map[string]string
+	if len(r.Labels) > 0 {
```

Comment:
That's an API design issue (probably).  Replication controllers do it via
defaulting.  We should probably do it via defaulting.  You don't need to do
it here though because we do need to merge.  So this is LGTM barring the
error condition.

On Thu, Jul 23, 2015 at 5:20 PM, Cesar Wong <notifications@github.com>
wrote:

> In pkg/generate/app/app.go
> <https://github.com/openshift/origin/pull/3882#discussion_r35372958>:
>
> > @@ -431,8 +432,13 @@ func (r *DeploymentConfigRef) DeploymentConfig() (*deployapi.DeploymentConfig, e
> >  		r.Name = name
> >  	}
> >
> > -	selector := map[string]string{
> > -		"deploymentconfig": r.Name,
> > +	var selector map[string]string
> > +	if len(r.Labels) > 0 {
>
> Ahh... got it. (Sorry I'm slow :-) ) Where do we do that? Do we do it in
> conversion or as a defaulting function for the dc?
>
> And why would we want to do that? Isn't it enough to explicitly set it in
> new-app ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/3882/files#r35372958>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift